File: leetcode/C++/remove-duplicates-from-sorted-array-ii.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        const int MAX_DUPLICATION = 2;
        int idx = 0, cnt = 1;
        for(int i=1; i<nums.size(); ++i){
            if(nums[idx]==nums[i]){if(++cnt<=MAX_DUPLICATION) nums[++idx]=nums[i];}
            else {nums[++idx] = nums[i]; cnt = 1;}
        }
        return idx + 1;
    }
};

File: leetcode/Python/remove-duplicates-from-sorted-array-ii.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        MAX_DUPLICATION = 2
        cnt, idx = 1, 0
        for i in range(1, len(nums)):
            if nums[idx] == nums[i]:
                if cnt < MAX_DUPLICATION:
                    idx = idx + 1
                    nums[idx] = nums[i]
                cnt += 1
            else:
                idx, cnt = idx + 1, 1
                nums[idx] = nums[i]
        return idx + 1
                    